game-core/src/main/java/games/strategy/triplea/ui/mapdata/MapData.java

```diff
@@ -337,6 +346,68 @@ public int getDefaultUnitsStackSize() {
     return Math.max(0, Integer.parseInt(stack));
   }
 
+  /** Returns the unit color associated with the player named {@code playerName}. */
+  public Optional<Color> getUnitColor(final String playerName) {
+    // already loaded, just return
+    if (unitColors.containsKey(playerName)) {
+      return Optional.ofNullable(unitColors.get(playerName));
+    }
+    // look in map.properties
+    final Color color;
+    try {
+      color = getColorProperty(PROPERTY_UNITS_COLOR_PREFIX + playerName);
+    } catch (final Exception e) {
```

Comment:
Nit, catching raw Exception can be considered a code smell. If we have a bug somewhere and throw a NPE, the error message might be incorrect. If we're going to throw an exception anyways, would we be just as well off if `getColorProperty` threw that exception (unchecked)? In this case this method would not need a try/catch block ata ll.

Reply:
Yeah, the exception handling in a lot of this class is questionable. I had followed a similar pattern as playerColors but decided to go ahead and refactor this a bit to clean it up.